### ISSUE TITLE
Support SOCKS5_PROXY and HTTP_PROXY for transport

### ISSUE THREAD

**Issue by rapiz1** [rapiz1/rathole#35]
Like:

```toml
[client.transport]
type = "tcp"
[client.transport.tcp]
proxy = "socks5://localhost:1080" # connect to the server via a socks5 proxy
```

or
```toml
[client.transport]
type = "tcp"
[client.transport.tcp]
proxy = "http://localhost:1080" # connect to the server via a http proxy
```

**Comment by leew66884** [comment]
I also need this plugin！！！

**Comment by shijunti19** [comment]
I also need this plugin！！！

**Comment by rapiz1** [comment]
It's the next feature I'm going to work on. Just please be patient :)



---Original---
From: ***@***.***&gt;
Date: Tue, Feb 22, 2022 14:59 PM
To: ***@***.***&gt;;
Cc: "Yujia ***@***.******@***.***&gt;;
Subject: Re: [rapiz1/rathole] Support SOCKS5_PROXY and HTTP_PROXY fortransport (Issue #35)




 
I also need this plugin！！！

**Comment by darkr4y** [comment]
Is it possible to implement a client plugin like frp,not only through a proxy to access the rathole server
but like this https://github.com/fatedier/frp#client-plugins
```
# frpc.ini
[plugin_socks5]
type = tcp
remote_port = 6005
plugin = socks5
plugin_user = abc
plugin_passwd = abc
```
Expose a socks5 proxy service to the external network

**Comment by rapiz1** [comment]
Released in v0.4.0
